subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/FileSystemWatchingBuildActionRunner.java

```diff
@@ -44,8 +48,12 @@ public FileSystemWatchingBuildActionRunner(BuildActionRunner delegate) {
     public Result run(BuildAction action, BuildController buildController) {
         GradleInternal gradle = buildController.getGradle();
         StartParameterInternal startParameter = gradle.getStartParameter();
-        BuildLifecycleAwareVirtualFileSystem virtualFileSystem = gradle.getServices().get(BuildLifecycleAwareVirtualFileSystem.class);
-        BuildOperationRunner buildOperationRunner = gradle.getServices().get(BuildOperationRunner.class);
+        ServiceRegistry services = gradle.getServices();
+        BuildLifecycleAwareVirtualFileSystem virtualFileSystem = services.get(BuildLifecycleAwareVirtualFileSystem.class);
```

Comment:
Should we reset the statistics when the build starts so we don't capture any statistics of things which happened between builds?

If I understand correctly, the current code also captures statistics from previous builds which do not have the debug flag enabled, since we don't reset the statistics at the beginning or the end.

Reply:
That's a good point about the flag, that should be fixed.

As to events between builds I think the right solution would be what we do with the VFS events, i.e. to print the statistics before the build, too.

Reply:
So let's print the statistics at the beginning and at the end of the build when the flag is present.

Reply:
Done.